modules/mono/glue/runtime_interop.cpp

```diff
@@ -304,6 +312,19 @@ void godotsharp_internal_new_csharp_script(Ref<CSharpScript> *r_dest) {
 	memnew_placement(r_dest, Ref<CSharpScript>(memnew(CSharpScript)));
 }
 
+void godotsharp_internal_editor_file_system_update_file(const String *p_script_path) {
+#if TOOLS_ENABLED
+	// If the EditorFileSystem singleton is available, update the file;
+	// otherwise, the file will be updated when the singleton becomes available.
+	EditorFileSystem *efs = EditorFileSystem::get_singleton();
+	if (efs) {
+		efs->update_file(*p_script_path);
+	}
+#else
+	CRASH_NOW_MSG("EditorFileSystem is only available when running in the Godot editor.");
```

Comment:
We could add a `DEV_ASSERT`, but for production user templates is it really worth crashing instead of just erroring out? Or is it impossible to recover once we're here?

Reply:
It is only called behind a IsEditorHint check, so it should be unreachable unless someone messes around with reflection (except that the check would have crashed again, so good that you commented and I double checked)

Thb calling this outside the editor means just that there is a logic error in the caller, so a DEV_ASSERT or just a normal ERR_FAIL would be enough here.

Reply:
It is as RedworkDE says, this code path should be unreachable so I used a crash because a user should never see it anyway. But it makes sense to me to use a `DEV_ASSERT` too so I changed it. Let me know if the check is backwards.